src/tools/miri/tests/fail/both_borrows/buggy_as_mut_slice.rs

```diff
@@ -13,7 +13,7 @@ fn main() {
     let v1 = safe::as_mut_slice(&v);
     let v2 = safe::as_mut_slice(&v);
     v1[1] = 5;
-    //~[stack]^ ERROR: /write access .* tag does not exist in the borrow stack/
+    //~[stack]^ ERROR: /trying to retag .+ for SharedReadOnly permission .+ tag does not exist in the borrow stack for this location/
```

Comment:
Hm, not seems to fail on a read / retag-for-read now, instead of a write. That is a bit surprising. How did the MIR change here?

Reply:
Ah, yeah the `&raw const` here is relevant for Stacked Borrows -- it basically acts as a read on the entire slice. That is a bit unfortunate since so far, for slice accesses, we didn't do such a whole-slice retag. I think ideally we would not add a retag to these raw reborrows.

Is there any way for the retag pass to recognize these reborrows? Like, are they marked as syntactic sugar somewhere, or so? I recall we do that for some constructs, but I am not actually sure how it is represented.

Long-term I hope we can change SB to make raw retags true NOPs, but that will require a bit more work.

Reply:
I'm afraid I don't know much about MIR borrowck.

Reply:
> I guess if I was to mark the &raw const as something it'd be in the SourceInfo that I'd do that, by marking its span somehow?

Yeah I think that's how the desugaring markers usually work, but I am not sure.